www/class/centreonTraps.class.php

```diff
@@ -284,135 +284,216 @@ public function update($traps_id = null)
 
         $rq = "UPDATE traps SET ";
         $rq .= "`traps_name` = ";
-        isset($ret["traps_name"]) && $ret["traps_name"] != null
-            ? $rq .= "'" . $ret["traps_name"] . "', "
-            : $rq .= "NULL, ";
+        if (isset($ret["traps_name"]) && $ret["traps_name"] != null) {
+            $rq .= ':traps_name, ';
+            $retValue[':traps_name'] = $ret["traps_name"];
+        } else {
+            $rq .= "NULL, ";
+        }
 
         $rq .= "`traps_oid` = ";
-        isset($ret["traps_oid"]) && $ret["traps_oid"] != null
-            ? $rq .= "'" . $ret["traps_oid"] . "', "
-            : $rq .= "NULL, ";
+        if (isset($ret["traps_oid"]) && $ret["traps_oid"] != null) {
+            $rq .= ':traps_oid, ';
+            $retValue[':traps_oid'] = $ret["traps_oid"];
+        } else {
```

Comment:
as this value is mandatory, you're not able to save the form and this case shouldn't exist